core/arch/arm/kernel/entry_a64.S

```diff
@@ -137,9 +137,8 @@
 	.endm
 
 	.macro init_pauth_per_cpu
-		msr	spsel, #1
-		ldp	x0, x1, [sp, #THREAD_CORE_LOCAL_KEYS]
-		msr	spsel, #0
+		adr_l	x3, threads
```

Comment:
In this case, I reworked this commit and I have a different approach of the problem. Instead of trying to change keys/context I tell the compiler with attribute to NOT compile with pointer authentication the functions which are affected the changing of the keys. This way the feature is already enabled when the SP is loaded, but some functions in optee core will not contain the paciasp/autiasp instructions avoiding the errors.
What is your opinion of this solution?

Reply:
I'm not so keen on this approach. It's messy and confusing with the `__pauth` directives. I'd rather update the runtime environment to be ready for pointer authentication.

Reply:
By "ready for pointer authentication" do you mean initializing the kernel thread and switching to it right before the call of call_finalcalls instead of using the atomic thread ?